packages/libcanberra.rb

```diff
@@ -3,36 +3,105 @@
+            # gtk: Don't assume all GdkDisplays are GdkX11Displays: broadway/wayland
+      #{'      '}
+            --- src/canberra-gtk-module.c.orig
+            +++ src/canberra-gtk-module.c
+            @@ -307,6 +307,11 @@ static gint window_get_desktop(GdkDisplay *d, GdkWindow *w) {
+                     guchar *data = NULL;
+                     gint ret = -1;
+      #{'       '}
+            +#ifdef GDK_IS_X11_DISPLAY
+            +        if (!GDK_IS_X11_DISPLAY(d))
+            +                return 0;
+            +#endif
+            +
+                     if (XGetWindowProperty(GDK_DISPLAY_XDISPLAY(d), GDK_WINDOW_XID(w),
+                                            gdk_x11_get_xatom_by_name_for_display(d, "_NET_WM_DESKTOP"),
+                                            0, G_MAXLONG, False, XA_CARDINAL, &type_return,
+            @@ -335,6 +340,11 @@ static gint display_get_desktop(GdkDisplay *d) {
+                     guchar *data = NULL;
+                     gint ret = -1;
+      #{'       '}
+            +#ifdef GDK_IS_X11_DISPLAY
+            +        if (!GDK_IS_X11_DISPLAY(d))
+            +                return 0;
+            +#endif
+            +
+                     if (XGetWindowProperty(GDK_DISPLAY_XDISPLAY(d), DefaultRootWindow(GDK_DISPLAY_XDISPLAY(d)),
+                                            gdk_x11_get_xatom_by_name_for_display(d, "_NET_CURRENT_DESKTOP"),
+                                            0, G_MAXLONG, False, XA_CARDINAL, &type_return,
+            @@ -365,6 +375,11 @@ static gboolean window_is_xembed(GdkDisplay *d, GdkWindow *w) {
+                     gboolean ret = FALSE;
+                     Atom xembed;
+      #{'       '}
+            +#ifdef GDK_IS_X11_DISPLAY
+            +        if (!GDK_IS_X11_DISPLAY(d))
+            +                return FALSE;
+            +#endif
+            +
+                     /* Gnome Panel applets are XEMBED windows. We need to make sure we
+                      * ignore them */
+      #{'       '}
+            --- src/canberra-gtk.c.orig
+            +++ src/canberra-gtk.c
+            @@ -185,6 +185,11 @@ static gint window_get_desktop(GdkDisplay *d, GdkWindow *w) {
+                     guchar *data = NULL;
+                     gint ret = -1;
+      #{'       '}
+            +#ifdef GDK_IS_X11_DISPLAY
+            +        if (!GDK_IS_X11_DISPLAY(d))
+            +                return 0;
+            +#endif
+            +
+                     if (XGetWindowProperty(GDK_DISPLAY_XDISPLAY(d), GDK_WINDOW_XID(w),
+                                            gdk_x11_get_xatom_by_name_for_display(d, "_NET_WM_DESKTOP"),
+                                            0, G_MAXLONG, False, XA_CARDINAL, &type_return,
+    LIBCANBERRA_WAYLAND_PATCH_HEREDOC
+    IO.write('libcanberra-wayland-crash.patch', @libcanberra_wayland_crash_patch, perm: 0o555)
+    system 'patch -p0 < libcanberra-wayland-crash.patch'
```

Comment:
Epic patch!

Reply:
Yeah, but it's a package which hasn't been updated in 8 years... And the patch was mentioned on a mailing list, so I figured that we should just add it to the package instead of pulling it from somewhere where the source might get lost.